notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala

```diff
@@ -115,25 +113,47 @@ trait HarvesterRequestHandler extends Logging {
         .unsafeRunSync()
     }catch {
       case e: Throwable => {
-        records.foreach(record =>
-          logger.error(Map(
-            "notificationId" -> record.notification.id,
-            "notificationType" -> record.notification.`type`.toString,
-          ), s"Error occurred: ${e.getMessage}", e)
-        )
+        records.foreach {
+          case (body, _) =>
+            logger.error(Map(
+              "notificationId" -> body.notification.id,
+              "notificationType" -> body.notification.`type`.toString,
+            ), s"Error occurred: ${e.getMessage}", e)
+        }
         throw e
       }
     }finally {
-      val end = Instant.now
-      records.foreach(record =>
-        logger.info(Map(
-          "notificationId" -> record.notification.id,
-          "notificationType" -> record.notification.`type`.toString,
-          "harvester.notificationProcessingTime" -> Duration.between(start, end).toMillis,
-          "harvester.notificationProcessingEndTime.millis" -> end.toEpochMilli,
-          "harvester.notificationProcessingEndTime.string" -> end.toString,
-        ), "Finished processing notification event")
-      )
+      records.foreach {
```

Comment:
At the moment the harvester batchSize is 1. I think if we increased this then logging in the `finally` block won't provide us with accurate metrics. Not something that I _think_ I need to address now, just a consideration for the future